File: Linkpoint/src/main/java/com/linkpoint/render/spatial/DrawEntryList.java
package com.linkpoint.render.spatial;
import java.util.*;

import com.linkpoint.utils.InlineList;
import javax.annotation.Nullable;

public class DrawEntryList extends InlineList<DrawListEntry> {
    @Nullable
    private final EntryRemovalListener listener;

    public interface EntryRemovalListener {
        void onEntryRemovalRequested(DrawListEntry drawListEntry);
    }

    public DrawEntryList(@Nullable EntryRemovalListener entryRemovalListener) {
        this.listener = entryRemovalListener;
    }

    public void requestEntryRemoval(DrawListEntry drawListEntry) {
        if (this.listener != null) {
            this.listener.onEntryRemovalRequested(drawListEntry);
        }
    }
}

File: Linkpoint/src/main/app/com/linkpoint/render/spatial/DrawEntryList.kt
package com.linkpoint.render.spatial

import com.linkpoint.utils.InlineList

class DrawEntryList(private val listener: EntryRemovalListener?) : InlineList<DrawListEntry>() {
    interface EntryRemovalListener {
        fun onEntryRemovalRequested(drawListEntry: DrawListEntry)
    }

    override fun requestEntryRemoval(drawListEntry: DrawListEntry) {
        listener?.onEntryRemovalRequested(drawListEntry)
    }
}